src/go/rpk/pkg/cli/container/common/common.go

```diff
@@ -506,3 +611,41 @@ const ContainerProfileName = "rpk-container"
 // ErrContainerProfileExists is returned when we attempt to create a container
 // profile but a profile named 'rpk-container' already exists.
 var ErrContainerProfileExists = fmt.Errorf("%q profile already exists", ContainerProfileName)
+
+func parseConsoleConfigFile(kafkaAddr, srAddr, adminAddr []string) (string, error) {
+	type Listener struct {
+		Enabled bool     `yaml:"enabled"`
+		Urls    []string `yaml:"urls"`
+	}
+	type Kafka struct {
+		Brokers        []string `yaml:"brokers"`
+		SchemaRegistry Listener `yaml:"schemaRegistry"`
+	}
+	type Redpanda struct {
+		AdminAPI Listener `yaml:"adminApi"`
+	}
+	type ConsoleCfg struct {
+		Kafka    Kafka    `yaml:"kafka"`
+		Redpanda Redpanda `yaml:"redpanda"`
+	}
```

Comment:
personal nit but I'd merge these into one nested type struct for better symmetry with the definition just below